make.ps1

```diff
@@ -115,39 +116,97 @@ function GenerateRunSettings([String] $framework, [String] $platform, [String] $
+                Write-Host "$testName succeeded"
+            }
+
+            $bgJobs.Remove($finished)
+            Remove-Job $finished
+            if ($nextJob -lt $tasks.Count) {
+                $j = Start-Job -ScriptBlock $testScript -ArgumentList $tasks[$nextJob].params
+                $bgJobs[$j] = $nextJob;
+                $nextJob += 1
+            }
+            Write-Host "$($bgJobs.Count) jobs running: $(($bgJobs.values | ForEach-Object {$tasks[$_].name}) -Join ", ")"
+            Write-Host "$($tasks.Count - $nextJob) jobs pending: $(($nextJob..$tasks.Count | ForEach-Object {$tasks[$_].name}) -Join ", ")"
+            Write-Host
+        }
+    }
+}
+
 function Test([String] $target, [String] $configuration, [String[]] $frameworks, [String] $platform) {
-    foreach ($framework in $frameworks) {
-        $testname = "";
-        $filtername = $target
+    Write-Host "Scheduling Test Tasks"
+    Write-Host
 
+    $tasks = @()
+    foreach ($framework in $frameworks) {
         # generate the runsettings file for the settings
         $runSettings = GenerateRunSettings $framework $platform $configuration $runIgnored
 
+        function createTask($filtername, $filter) {
+            [Object[]] $args = @("$_BASEDIR/Src/IronPythonTest/IronPythonTest.csproj", '-f', "$framework", '-o', "$_BASEDIR/bin/$configuration/$framework", '-c', "$configuration", '--no-build', '-l', "trx;LogFileName=$filtername-$framework-$configuration-result.trx", '-s', "$runSettings", "--filter=$filter");
+            Write-Host "Enqueue [$framework $filtername]:"
+            Write-Host "dotnet test $args"
+            Write-Host
+            return @{ name = "[$framework $filtername]"; params = $args }
+        }
+
         $filter = $target
 
         switch ($filter.ToLower()) {
-            "all"        { $filter = ""}
-            "ironpython" { $filter = "TestCategory=IronPython" }
-            "cpython"    { $filter = "TestCategory=CPython" }
+            "all"        {
+                $tasks += createTask "ironpython" "TestCategory=IronPython"
+                $tasks += createTask "cpython" "TestCategory=CPython"
+                $tasks += createTask "cs" "TestCategory!=IronPython&TestCategory!=CPython"
+            }
+            "ironpython" { $tasks += createTask "ironpython" "TestCategory=IronPython" }
+            "cpython"    { $tasks += createTask "cpython" "TestCategory=CPython" }
             default      {
-                $filtername = "query"
-                if ($filter.ToLower().StartsWith('ironpython')) { $filter = "TestCategory=IronPython&" + $filter }
-                if ($filter.ToLower().StartsWith('cpython')) { $filter = "TestCategory=CPython&" + $filter }
+                if ($filter.ToLower().StartsWith('ironpython')) { $tasks += createTask "query" "TestCategory=IronPython&$filter" }
+                if ($filter.ToLower().StartsWith('cpython')) { $tasks += createTask "query" "TestCategory=CPython&$filter" }
```

Comment:
Looks like this breaks `test-<anything>` to do specific queries. Probably need to change to `elseif` and add:
```
else { $tasks += createTask "query" "$filter" }
```

Reply:
You're right! Fixed.